### ISSUE TITLE
Bug: useDeferredValue not respecting timeoutMs parameter

### ISSUE THREAD

**Issue by lourd** [facebook/react#19998]
React version: 0.0.0-experimental-4ead6b530
ReactDOM version: 0.0.0-experimental-4ead6b530

## Steps To Reproduce

I'm not sure if this is a bug or a misunderstanding. As part of learning Suspense and using it in a project, I was puzzled by the behavior of useDeferredValue. When using it, my component seemed to always use the deferred value regardless of the `timeoutMs` parameter. 

When using useDeferredValue in the child component, the `timeoutMs` does have the expected effect, in either version. But I forked the [Code Sandbox linked](https://codesandbox.io/s/vigorous-keller-3ed2b) from the [Concurrent UI Patterns](https://reactjs.org/docs/concurrent-mode-patterns.html#deferring-a-value) to compare, and moved the use of `useDeferredValue` higher to the `App` component to match how I was using it.

In [`0.0.0-experimental-5faf377df`](https://codesandbox.io/s/vibrant-mayer-jsock?file=/src/index.js:364-443), it does have an effect. But in the most recent `react@experimental`, [`0.0.0-experimental-4ead6b530`](https://codesandbox.io/s/peaceful-mestorf-70i8l?file=/src/index.js:422-501), it no longer has an effect.

```js
function App() {
  const [resource, setResource] = useState(initialResource);
  const deferredResource = useDeferredValue(resource, {
    timeoutMs: 42
  });
  const isPending = deferredResource !== resource;
  return (
    <>
      <button
        disabled={isPending}
        onClick={() => {
          // startTransition(() => {
          const nextUserId = getNextId(resource.userId);
          setResource(fetchProfileData(nextUserId));
          // });
        }}
      >
        Next
      </button>
      {isPending ? " Loading..." : null}
      <ProfilePage
        resource={deferredResource}
        isStale={deferredResource !== resource}
      />
    </>
  );
}

function ProfilePage({ resource, isStale }) {
  return (
    <Suspense fallback={<h1>Loading profile...</h1>}>
      <ProfileDetails resource={resource} />
      <Suspense fallback={<h1>Loading posts...</h1>}>
        <ProfileTimeline resource={resource} isStale={isStale} />
      </Suspense>
    </Suspense>
  );
}
```

**Comment by threepointone** [comment]
This is by design https://github.com/facebook/react/pull/19703 https://github.com/facebook/react/pull/19704

**Comment by gaearon** [comment]
Yeah the docs unfortunately are getting outdated. We’ll need to do a cleanup pass.

**Comment by lourd** [comment]
Gotcha, thanks for the links and clarification!
